tests/lib/rules/highest-available-document-mode/tests.ts

```diff
@@ -91,25 +91,46 @@ const testsForRequireMetaTagConfig: Array<IRuleTest> = [
     }
 ];
 
-const testsForTargetBrowsersConfig: Array<IRuleTest> = [
+const testsForHeaderWithBlankTargetBrowsersConfig: Array<IRuleTest> = [
     {
-        name: `HTML page is served with 'X-UA-Compatible' header but the targeted browsers don't support document modes`,
-        reports: [{ message: `'x-ua-compatible' header is not needed` }],
+        name: `HTML page is served without the 'X-UA-Compatible' header, but it's required by the targeted browsers`,
+        reports: [{ message: `'x-ua-compatible' header was not specified` }],
+        serverConfig: { '/': '' }
+    }
+];
+
+const testsForMetaTagWithBlankTargetBrowsersConfig: Array<IRuleTest> = [
+    {
+        name: `'X-UA-Compatible' meta tag is not specified, but it's required by the targeted browsers`,
+        reports: [{ message: `No 'x-ua-compatible' meta tag was specified` }],
         serverConfig: { '/': { headers: { 'X-UA-Compatible': 'ie=edge' } } }
-    },
+    }
+];
+
+const testsForTargetBrowsersConfig: Array<IRuleTest> = [
     {
-        name: `'X-UA-Compatible' meta tag is not specified but the targeted browsers don't support document modes`,
-        reports: [{ message: `Meta tag is not needed` }],
-        serverConfig: { '/': { content: generateHTMLPageWithMetaTag() } }
+        name: `HTML page is served with both 'X-UA-Compatible' header and meta tag, but they are not required by the targeted browsers`,
+        reports: [
+            { message: `'x-ua-compatible' header is not needed` },
+            { message: `Meta tag is not needed` }
+        ],
+        serverConfig: {
+            '/': {
+                content: generateHTMLPageWithMetaTag(),
+                headers: { 'X-UA-Compatible': 'ie=edge' }
+            }
+        }
     }
 ];
 
 ruleRunner.testRule(ruleName, testsForDefaults);
 ruleRunner.testRule(ruleName, testsForRequireMetaTagConfig, { ruleOptions: { requireMetaTag: true } });
+ruleRunner.testRule(ruleName, testsForHeaderWithBlankTargetBrowsersConfig, { browserslist: [] });
+ruleRunner.testRule(ruleName, testsForMetaTagWithBlankTargetBrowsersConfig, {
+    browserslist: [],
```

Comment:
The behavior when `browserslist` is not set is different from `browserslist` is set as empty. See https://github.com/sonarwhal/sonar/issues/446#issuecomment-324388133. It doesn't affect the result of this test though.